posthog/ai/utils.py

```diff
@@ -0,0 +1,156 @@
+import time
+from typing import Any, Callable, Dict, Optional
+
+from posthog.client import Client as PostHogClient
+
+
+def get_model_params(kwargs: Dict[str, Any]) -> Dict[str, Any]:
+    """
+    Extracts model parameters from the kwargs dictionary.
+    """
+    model_params = {}
+    for param in [
+        "temperature",
+        "max_tokens",
+        "top_p",
+        "frequency_penalty",
+        "presence_penalty",
+        "n",
+        "stop",
+        "stream",
+    ]:
+        if param in kwargs:
+            model_params[param] = kwargs.get(param)
```

Comment:
Do we want to send null values here? I would get rid of them.

Reply:
this is what this does right? only adds them if it has a value

Reply:
Yeah, so actually this would be slightly more Pythonic given the `if` above:
```suggestion
            model_params[param] = kwargs[param]
```